onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py

```diff
@@ -150,7 +210,34 @@ def chain_model(args):
         if args.precision == Precision.FLOAT16
         else [node]
     )
+    if args.output_no_speech_probs:
+        prob_cast_node = helper.make_node(
+            "Cast",
+            inputs=["no_speech_probs_beam"],
+            outputs=["no_speech_probs"],
+            name="no_speech_probs_cast_to_fp32",
+            to=TensorProto.FLOAT,
+        )
+        graph_nodes.extend([prob_cast_node])
+
     beam_graph = helper.make_graph(graph_nodes, "beam-search-test", graph_inputs, graph_outputs, initializers)
+    beam_graph_input_names = [gi.name for gi in graph_inputs]
+    beam_graph_output_names = [go.name for go in graph_outputs]
+
+    if args.cross_qk_onnx_model:
+        post_qk_model = onnx.load_model(args.cross_qk_onnx_model, load_external_data=True)
+        post_qk_graph = post_qk_model.graph
+        beam_graph.initializer.extend(post_qk_graph.initializer)
+        beam_graph.node.extend(post_qk_graph.node)
+        # TODO: Treat same name same input, user need check their shapes, etc
```

Comment:
We check that the necessary graph inputs are in the model in `verify_inputs`. You can add more checks there if needed.
https://github.com/microsoft/onnxruntime/blob/f299016cbe87a5341e0a8aa69b621555c9d49a35/onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py#L13-L19

Reply:
We people want to chain it, we could suppose they know the graph of both well. Checking may limit their works.